lxd/util/sys.go

```diff
@@ -110,3 +110,14 @@ func AddFileDescriptor(fdFiles *[]*os.File, file *os.File) int {
 	*fdFiles = append(*fdFiles, file)
 	return 2 + len(*fdFiles) // Use 2+fdFiles count, as first user file descriptor is 3.
 }
+
+// ShortenedSocketPath creates a shorter alternative path to a socket by using the file descriptor to the directory of the socket file.
```

Comment:
Can this function only be used for unix sockets, or can it be used for other types of files?

If the former, then it would make sense to include "Unix" in the function description at least.